src/workerd/jsg/setup.c++

```diff
@@ -562,6 +564,21 @@ void IsolateBase::jitCodeEvent(const v8::JitCodeEvent* event) noexcept {
   }
 }
 
+void* getJsCageBase() {
+  if (!v8Initialized) return nullptr;
+  v8::Isolate* isolate = v8::Isolate::TryGetCurrent();
+  if (isolate == nullptr) return nullptr;
+  // Returns null if setJsCageBase was never called.
+  return isolate->GetData(SET_DATA_CAGE_BASE);
+}
+
+void setJsCageBase(void* cageBase) {
```

Comment:
[nit] Unless you plan to call this from lots of different call sites it seems like these methods mainly serve to silently hide subtle, presumably unexpected error cases (the early returns). Maybe it would be better to just not have these?

Reply:
It's a layering issue - Edgeworker doesn't have a lot of files that deal directly with the v8 Isolate, so I thought it made more sense to put it in jsg. The early returns are modelled on getJsStackTrace, and like that function the getJsCageBase is intended to be called during crash handling so it is written in a defensive style.